src/Analyser/TypeSpecifier.php

```diff
@@ -674,59 +675,75 @@ public function specifyTypesInCondition(
 			&& count($expr->vars) > 0
 			&& !$context->null()
 		) {
-			if (!$context->true()) {
-				if (!$scope instanceof MutatingScope) {
-					throw new ShouldNotHappenException();
-				}
-
-				$specifiedTypes = new SpecifiedTypes();
+			// rewrite multi param isset() to and-chained single param isset()
+			if (count($expr->vars) > 1) {
+				$issets = [];
 				foreach ($expr->vars as $var) {
-					$isset = $scope->issetCheck($var, static fn () => true);
+					$issets[] = new Expr\Isset_([$var], $expr->getAttributes());
+				}
 
-					if ($isset !== true) {
+				$first = array_shift($issets);
+				$andChain = null;
+				foreach ($issets as $isset) {
+					if ($andChain === null) {
+						$andChain = new BooleanAnd($first, $isset);
 						continue;
 					}
 
-					$specifiedTypes = $specifiedTypes->unionWith($this->create(
-						$var,
-						new NullType(),
-						$context->negate(),
-						false,
-						$scope,
-						$rootExpr,
-					));
+					$andChain = new BooleanAnd($andChain, $isset);
 				}
 
-				return $specifiedTypes;
+				if ($andChain === null) {
+					throw new ShouldNotHappenException();
+				}
+
+				return $this->specifyTypesInCondition($scope, $andChain, $context, $rootExpr);
 			}
 
-			$vars = [];
-			foreach ($expr->vars as $var) {
-				$tmpVars = [$var];
+			$issetExpr = $expr->vars[0];
 
-				while (
-					$var instanceof ArrayDimFetch
-					|| $var instanceof PropertyFetch
-					|| (
-						$var instanceof StaticPropertyFetch
-						&& $var->class instanceof Expr
-					)
-				) {
-					if ($var instanceof StaticPropertyFetch) {
-						/** @var Expr $var */
-						$var = $var->class;
-					} else {
-						$var = $var->var;
-					}
-					$tmpVars[] = $var;
+			if (!$context->true()) {
+				if (!$scope instanceof MutatingScope) {
+					throw new ShouldNotHappenException();
+				}
+
+				$isset = $scope->issetCheck($issetExpr, static fn () => true);
+
+				if ($isset !== true) {
+					return new SpecifiedTypes();
 				}
 
-				$vars = array_merge($vars, array_reverse($tmpVars));
+				return $this->create(
+					$issetExpr,
+					new NullType(),
+					$context->negate(),
+					false,
+					$scope,
+					$rootExpr,
+				);
 			}
 
-			$types = null;
+			$tmpVars = [$issetExpr];
```

Comment:
I had to rename this var, because the previous `$var` variable after this PR is no longer within a foreach and therefore PHPStan would otherwise report a overlap with the foreach-var below